core/types/calls.cc

```diff
@@ -2529,25 +2529,25 @@ class Magic_callWithBlockPass : public IntrinsicMethod {
     friend class Magic_callWithSplatAndBlockPass;
 
 private:
-    static TypePtr typeToProc(const GlobalState &gs, const TypeAndOrigins &blockType, core::FileRef file,
-                              LocOffsets callLoc, LocOffsets receiverLoc, LocOffsets funLoc, Loc originForUninitialized,
-                              bool suppressErrors) {
+    static TypeAndOrigins typeToProc(const GlobalState &gs, const TypeAndOrigins &blockType, core::FileRef file,
+                                     LocOffsets callLoc, LocOffsets receiverLoc, LocOffsets funLoc,
+                                     Loc originForUninitialized, bool suppressErrors) {
```

Comment:
I have FUD here about copying an entire `TypeAndOrigins` that's really only relevant in the error case -- so we're making every call-with-block more expensive to show better errors in the rare case that there is an error.

We could instead keep this function the same and make sure that we pass in (span or ref of) the locs for the block to `simulateCall`, so that we can reconstruct a `TypeAndOrigins` for the error case?

(I don't have a good sense of how often we call through this code during a typical typecheck, just trying to avoid the accumulation of small inefficiencies along the way.)

Reply:
I backed it out to threading an origins vector through instead of constructing it eagerly.